Optionally read database target from check metadata

mrichar1 opened the pull request:
Hi,

Firstly, apologies in advance if you're not accepting PRs on this project - but having looked at the 17 forks of lusis' original project, yours seems to be the most stable and widely-used/contributed to...

If you are, then please consider this PR, which allows you to set the influx database to sent the metrics data on to by specifying a database in the check definition using the 'influxdb' key.  If you don't set one, the default 'database' key in the extension config is used instead.

Let me know what you think...

thanks!

masterkorp commented:
Hello @mrichar1, 

Yes we are, thank you for your Pull Request!
Can you please add a commit that updates the example on the README file too ?

mrichar1 commented:
Thanks for the proof-reading - documentation was never my strong point!  Have fixed the errors.

mrichar1 commented:
2 fixes to my previous commits:

1) Setting 'influxdb': 'db-name' limits the potential for future check-specific influxdb options. Modified to make influxdb a 'section' and use a 'database' option inside it.

2) My previous commit had wrong documentation, as I'd added the influxdb option to the handler config. Updated the docs to fix this, and to document the change in 1)

Since this is now a very messy series of commits and counter-commits, I'm happy to resubmit this PR just starting at the 'final' commit and ignore the various mistakes and fixes in the others if that would be useful?

masterkorp commented:
@mrichar1 Thank you for your changes.

Can you rebase that into the previous commits ?

``git rebase -i master``  and you can fixup the changes to the commit.

mrichar1 commented:
I think that 6609100 now contains a single commit with all the relevant changes - but please do check before pulling (rebase is always a little scary!)

masterkorp commented:
@mrichar1 Thanks 6609100 its correct.

But it seems that you have also rebase some commits in master, while the commit contents are the same as in our master branch, the hashes are now different.

To fix this you need delete these commits. But **backup to a branch** before tough with `git checkout -b backup`

```
git remote add upstream git@github.com:seegno/sensu-influxdb-extension.git
git remote update
git rebase -i 34b410c0a2
# Delete all the commits minus yours
git rebase upstream master
git push -f origin master
```

In the future I recommend you to work you a separated branch.

mrichar1 commented:
Argh - yeah, noticed that after I pushed, and was just reading up on how exacty remove them - thanks for the walkthrough!

I usually do work on a branch, but I was being lazy and thinking this would be a simple commit - won't make that mistake again!

fixe commented on influxdb.rb:
```diff
@@ -64,7 +65,8 @@ def parse_event(event_data)
             "series" => event["check"]["name"],
-            "timestamp" => event["check"]["issued"]
+            "timestamp" => event["check"]["issued"],
+            "database" => event["check"]["influxdb"]["database"]
```
Sort.

fixe commented on README.md:
```diff
@@ -46,6 +46,24 @@ Just drop the file in `/etc/sensu/extensions` and add it to your `metrics` confi
+      "command": "check-foo",
+      "handlers": ["metrics"],
+      "influxdb": {
+        "database": "other-db"
```
Rename `other-db` to `name`.

mrichar1 commented:
Amended commit as per comments from @fixe - thanks!
